### PR TITLE
cmake.eclass: New eclass

### PR DESCRIPTION
**Pull request by a17r**

### CODE REVIEW

File: eclass/cmake.eclass [hunk 1]
```diff
@@ -54,30 +52,19 @@ _CMAKE_UTILS_ECLASS=1
...
+# The default is set to "ninja".
+: ${CMAKE_MAKEFILE_GENERATOR:=ninja}
 
 # @ECLASS-VARIABLE: CMAKE_REMOVE_MODULES_LIST
 # @DESCRIPTION:
-# Space-separated list of CMake modules that will be removed in $S during src_prepare,
-# in order to force packages to use the system version.
+# Space-separated list of CMake modules that will be removed in $S during
+# src_prepare, in order to force packages to use the system version.
 : ${CMAKE_REMOVE_MODULES_LIST:=FindBLAS FindLAPACK}
```
Comment (mgorny) [inline eclass/cmake.eclass]: I'd add a comment like 'Set to empty to disable removing modules entirely'.

Comment (mgorny) [reply eclass/cmake.eclass]: Also, maybe add array support here?

File: eclass/cmake.eclass [hunk 2]
```diff
@@ -768,11 +583,11 @@ _cmake_emake_src_make() {
 
 }
 
-# @FUNCTION: cmake-utils_src_make
+# @FUNCTION: cmake_src_make
```
Comment (mgorny) [inline eclass/cmake.eclass]: To be honest, I always found the name `src_make` weird. Maybe `cmake_build`?

Comment (a17r) [reply eclass/cmake.eclass]: and then, likewise `_cmake_ninja_src_make`, `_cmake_emake_src_make` to `cmake_*_build`?

Comment (mgorny) [reply eclass/cmake.eclass]: I don't know. I don't think they need to be public API. In fact, I would personally just inline them in some little `case ... in`.

File: eclass/cmake.eclass [hunk 3]
```diff
@@ -768,25 +587,33 @@ _cmake_emake_src_make() {
...
 # All arguments are passed to emake.
-cmake-utils_src_make() {
+cmake_build() {
 	debug-print-function ${FUNCNAME} "$@"
 
 	_cmake_check_build_dir
 	pushd "${BUILD_DIR}" > /dev/null || die
 
-	_cmake_${CMAKE_MAKEFILE_GENERATOR}_src_make "$@"
+	_cmake_${CMAKE_MAKEFILE_GENERATOR}_build "$@"
```
Comment (junghans) [inline eclass/cmake.eclass]: Could we use `cmake --build .` here?

File: eclass/cmake.eclass [hunk 4]
```diff
@@ -819,15 +646,16 @@ cmake-utils_src_test() {
...
 # @DESCRIPTION:
 # Function for installing the package. Automatically detects the build type.
-cmake-utils_src_install() {
+cmake_src_install() {
 	debug-print-function ${FUNCNAME} "$@"
 
 	_cmake_check_build_dir
 	pushd "${BUILD_DIR}" > /dev/null || die
-	DESTDIR="${D}" ${CMAKE_MAKEFILE_GENERATOR} install "$@" || die "died running ${CMAKE_MAKEFILE_GENERATOR} install"
+	DESTDIR="${D}" ${CMAKE_MAKEFILE_GENERATOR} install "$@" ||
```
Comment (junghans) [inline eclass/cmake.eclass]: And `cmake --install` here.

File: eclass/cmake.eclass (hunk 3 above)
Comment (a17r) [reply eclass/cmake.eclass]: What's the benefit on giving up our established wrappers?

**Pull request merged**